The changes look good, thank you - I've got something similar in other projects where we have a build system in place for the wheels (see also #14 as I'd like to build at least Windows wheels for this backport).

Would you want to be acknowledged in the release notes (as "wiggin15")?

Would you want a new release with this soon?

Any idea why the Python nightly (since TravisCI does not yet have Python 3.7 available natively) build failed?

```
zip_safe flag not set; analyzing archive contents...
Traceback (most recent call last):
  File "setup.py", line 112, in <module>
    'build_ext': build_ext_subclass,
  File "/home/travis/virtualenv/python3.7-dev/lib/python3.7/site-packages/setuptools/__init__.py", line 129, in setup
    return distutils.core.setup(**attrs)
  File "/opt/python/3.7-dev/lib/python3.7/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/opt/python/3.7-dev/lib/python3.7/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/opt/python/3.7-dev/lib/python3.7/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/home/travis/virtualenv/python3.7-dev/lib/python3.7/site-packages/setuptools/command/install.py", line 67, in run
    self.do_egg_install()
  File "/home/travis/virtualenv/python3.7-dev/lib/python3.7/site-packages/setuptools/command/install.py", line 109, in do_egg_install
    self.run_command('bdist_egg')
  File "/opt/python/3.7-dev/lib/python3.7/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/opt/python/3.7-dev/lib/python3.7/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/home/travis/virtualenv/python3.7-dev/lib/python3.7/site-packages/setuptools/command/bdist_egg.py", line 218, in run
    os.path.join(archive_root, 'EGG-INFO'), self.zip_safe()
  File "/home/travis/virtualenv/python3.7-dev/lib/python3.7/site-packages/setuptools/command/bdist_egg.py", line 269, in zip_safe
    return analyze_egg(self.bdist_dir, self.stubs)
  File "/home/travis/virtualenv/python3.7-dev/lib/python3.7/site-packages/setuptools/command/bdist_egg.py", line 379, in analyze_egg
    safe = scan_module(egg_dir, base, name, stubs) and safe
  File "/home/travis/virtualenv/python3.7-dev/lib/python3.7/site-packages/setuptools/command/bdist_egg.py", line 416, in scan_module
    code = marshal.load(f)
ValueError: bad marshal data (unknown type code)
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -21,6 +21,9 @@ addons:
     packages:
     - liblzma-dev
 install:
+  # On nightly, upgrade setuptools first to work around
+  # https://github.com/pypa/setuptools/issues/1257
+  - if [[ $TRAVIS_PYTHON_VERSION == 'nightly' ]]; then travis_retry pip install -U setuptools; fi
   - "python setup.py install"
 before_script:
   - "cd test"

